### PR TITLE
AAP-18888 - Automated tests for Applications

### PR DESCRIPTION
**Pull request by acruzgon**
This PR is adding the automated tests for Applications and closes [AAP-18888](https://issues.redhat.com/browse/AAP-18888)

### CODE REVIEW

File: cypress/e2e/awx/administration/applications.cy.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  beforeEach(() => {
+    cy.awxLogin();
+    cy.navigateTo('awx', 'applications');
+  });
+
+  it('render the Applications page', () => {
+    cy.verifyPageTitle('OAuth Applications');
+  });
+
+  data.create_cases.forEach((item) => {
```
Comment (vishsanghishetty) [inline cypress/e2e/awx/administration/applications.cy.ts:14]: fixtures are usually used as data for component tests.

File: cypress/e2e/awx/administration/applications.cy.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const redirectUris = 'https://edit.com';
+
+    cy.createCustomAWXApplicationFromUI(
+      appName,
+      appDescription,
+      grantType,
+      clientType,
+      redirectUris
+    );
+    cy.DeleteCustomAWXApplicationFromListView(appName);
```
Comment (vishsanghishetty) [inline cypress/e2e/awx/administration/applications.cy.ts:65]: lowercase d, please

**Review by vishsanghishetty — APPROVED** [review]

**Review by akus062381 — APPROVED** [review]

File: cypress/e2e/awx/administration/applications.cy.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    const redirectUris = 'https://delete.com';
+
+    cy.createCustomAWXApplicationFromUI(
+      appName,
+      appDescription,
+      grantType,
+      clientType,
+      redirectUris
+    );
+    cy.DeleteCustomAWXApplicationFromDetailsView(appName, grantType, clientType);
```
Comment (vishsanghishetty) [inline cypress/e2e/awx/administration/applications.cy.ts:82]: same, lowercase d, please

File: cypress/e2e/awx/administration/applications.cy.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    cy.awxLogin();
+    cy.navigateTo('awx', 'applications');
+  });
+
+  it('render the Applications page', () => {
+    cy.verifyPageTitle('OAuth Applications');
+  });
+
+  data.create_cases.forEach((item) => {
+    it(`can create a single application with grant type ${item.auth_grant_type} based and client type ${item.client_type}`, function () {
```
Comment (vishsanghishetty) [inline cypress/e2e/awx/administration/applications.cy.ts:15]: create a custom command for Application creation via API, too

**Comment by vishsanghishetty** [comment]
@acruzgon looks like lint check has failed, if you see the checks above
please use `npm run prettier:fix`

File: cypress/support/commands.ts [hunk 5]
```diff
@@ -75,6 +75,35 @@ declare global {
...
 
+      createCustomAWXApplicationFromUI(
+        customAppName: string,
+        customAppDescription: string,
+        customGrantType: string,
+        customClientType: string,
+        customRedirectURIS: string
+      ): Chainable<void>;
+
+      EditCustomAWXApplicationFromDetailsView(
```
Comment (vishsanghishetty) [inline cypress/support/commands.ts:86]: lowercase for custom commands. we use Camel case, fyi

File: cypress/support/awx-commands.ts [hunk 6]
```diff
@@ -1232,3 +1242,179 @@ Cypress.Commands.add('createGlobalProject', function () {
...
+    cy.get('header').contains('Permanently delete applications');
+    cy.get('button').contains('Delete application').should('have.attr', 'aria-disabled', 'true');
+    cy.get('[data-cy="name-column-cell"]').should('have.text', customAppName);
+    cy.get('[data-cy="organization-column-cell"]').should('have.text', 'Default');
+    cy.get('input[id="confirm"]').click();
+    cy.get('button').contains('Delete application').click();
+  });
+
+  //Confirm status
+  cy.wait(200);
```
[the remark this answers is not in the document]
Comment (acruzgon) [reply cypress/support/awx-commands.ts:1373]: @vishsanghishetty progress bar to get to success is not immediate so the assertion was failing

**Pull request merged**